src/Microsoft.Tye.Hosting/PortAssigner.cs

```diff
@@ -50,7 +50,7 @@ static int GetNextPort()
                         binding.Port = GetNextPort();
                     }
 
-                    if (service.Description.Replicas == 1)
+                    if ((service.Description.Readiness is null || service.Description.RunInfo is IngressRunInfo) && service.Description.Replicas == 1)
```

Comment:
Why add the check for `service.Description.RunInfo is IngressRunInfo`

Reply:
because I forgot to remove when added `service.Description.Readiness is null` :)